packages/shell-api/src/shell-internal-state.ts

```diff
@@ -231,4 +239,94 @@ export default class ShellInternalState {
       }
     };
   }
+
+  async getDefaultPrompt(): Promise<string> {
+    if (this.cliOptions.nodb) {
+      return '> ';
+    }
+
+    if (typeof this.promptState.prefix === 'undefined') {
+      // since the connectionInfo is stable (unless we reconnect) we only compute it when missing
+      this.promptState.prefix = '';
+      try {
+        if (this.connectionInfo.buildInfo?.modules?.indexOf('enterprise') > -1) {
+          this.promptState.prefix += 'MongoDB Enterprise ';
+        }
+      } catch (e) {
+        // Shhhh... Did you see an error? I didn't...
+      }
+    }
+
+    let prompt = this.promptState.prefix;
+    if (!this.promptState.infoSource || this.promptState.infoSource === 'replSet') {
+      try {
+        prompt += await this.getPromptFromReplSet();
+        this.promptState.infoSource = 'replSet';
+      } catch (e) {
+        this.promptState.infoSource = 'master';
+      }
+    }
+    if (this.promptState.infoSource === 'master') {
+      try {
+        prompt += await this.getPromptFromIsMaster();
+      } catch (e) {
+        // out of options... we're not going to try again...
+        this.promptState.infoSource = 'unavailable';
+      }
+    }
+    return prompt + '> ';
+  }
+
+  private async getPromptFromReplSet(): Promise<string> {
+    let prompt = '';
+    // forShell suppresses error logging for theses commands on the server
+    const stateInfo = await this.currentDb.getSiblingDB('admin').runCommand({ replSetGetStatus: 1, forShell: 1 });
+    if (stateInfo.ok) {
+      // Report the self member's stateStr if it's present.
+      let state = stateInfo.members.find((m: any) => !!m.self)?.stateStr;
+      // Otherwise fall back to reporting the numeric myState field (mongodb 1.6).
+      if (!state) {
+        state = stateInfo.myState;
+      }
+      prompt = `${stateInfo.set}:${state}`;
+    } else {
+      const info = stateInfo.info;
+      if (info && info.length < 20) {
+        prompt = info; // "mongos", "configsvr"
+      } else {
+        throw new MongoshCommandFailed('replSetGetStatus failed');
+      }
+    }
+    return prompt;
+  }
+
+  private async getPromptFromIsMaster(): Promise<string> {
+    let prompt = '';
+    // forShell suppresses error logging for theses commands on the server
+    const isMaster = await this.currentDb.runCommand({ isMaster: 1, forShell: 1 });
+    if (isMaster.ok) {
+      let role = '';
+      if (isMaster.msg === 'isdbgrid') {
+        role = 'mongos';
+      }
+
+      if (isMaster.setName) {
+        if (isMaster.ismaster) {
+          role = 'PRIMARY';
+        } else if (isMaster.secondary) {
+          role = 'SECONDARY';
+        } else if (isMaster.arbiterOnly) {
+          role = 'ARBITER';
+        } else {
+          role = 'OTHER';
```

Comment:
Do we want to keep the uppercase names? That has a very 80s feel to it for me … :)

Reply:
:D I'm fine with either casing... would you suggest just having it lowercase?

Reply:
I think either lowercase or capitalized would be a bit nicer, but I don’t have a strong preference anyway :)